src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -80,25 +90,45 @@ public class ParameterProvider {
    *
    * @param parameterOverrides Map of parameter name to value
    * @param props Properties from profile file
+   * @param isIcebergMode If the provided parameters need to be verified and modified to meet
+   *     Iceberg mode
    */
-  public ParameterProvider(Map<String, Object> parameterOverrides, Properties props) {
-    this.setParameterMap(parameterOverrides, props);
+  public ParameterProvider(
+      Map<String, Object> parameterOverrides, Properties props, boolean isIcebergMode) {
+    this.setParameterMap(parameterOverrides, props, isIcebergMode);
   }
 
   /** Empty constructor for tests */
-  public ParameterProvider() {
-    this(null, null);
+  public ParameterProvider(boolean isIcebergMode) {
+    this(null, null, isIcebergMode);
   }
 
-  private void updateValue(
-      String key, Object defaultValue, Map<String, Object> parameterOverrides, Properties props) {
+  private void checkAndUpdate(
+      String key,
+      Object defaultValue,
+      Map<String, Object> parameterOverrides,
+      Properties props,
+      boolean enforceDefault) {
     if (parameterOverrides != null && props != null) {
       this.parameterMap.put(
           key, parameterOverrides.getOrDefault(key, props.getOrDefault(key, defaultValue)));
     } else if (parameterOverrides != null) {
       this.parameterMap.put(key, parameterOverrides.getOrDefault(key, defaultValue));
     } else if (props != null) {
       this.parameterMap.put(key, props.getOrDefault(key, defaultValue));
+    } else {
+      this.parameterMap.put(key, defaultValue);
+    }
+
+    if (enforceDefault) {
```

Comment:
please make sure there is a testcase to validate this codepath works as expected and you get an exception of the expected type.

Reply:
Added [here](https://github.com/snowflakedb/snowflake-ingest-java/blob/a37433d3c53bc442d28003b275ca50c755ca9ea2/src/test/java/net/snowflake/ingest/streaming/internal/ParameterProviderTest.java#L346C5-L352C16)